barretenberg/cpp/src/barretenberg/lmdblib/queries.cpp

```diff
@@ -23,7 +23,14 @@ void put_value(
     // The database has been configured to allow duplicate keys, but we don't permit duplicate key/value pairs
     // If we create a duplicate it will not insert it
     unsigned int flags = duplicatesPermitted ? MDB_NODUPDATA : 0U;
-    call_lmdb_func("mdb_put", mdb_put, tx.underlying(), db.underlying(), &dbKey, &dbVal, flags);
+    auto code = call_lmdb_func_with_return(mdb_put, tx.underlying(), db.underlying(), &dbKey, &dbVal, flags);
+    if (code == MDB_KEYEXIST && duplicatesPermitted) {
+        return;
+    }
```

Comment:
Pointing out this change since I encountered instances in this refactor where the same key/value entry was being added to a dupSort database.

This change makes the C++ module swallow the error making it behave more like the previous wrapper.